packages/odyssey-react-mui/src/labs/DataFilters.tsx

```diff
@@ -176,55 +176,49 @@ const DataFilters = ({
   }, [onChangeSearch, searchValue, searchDelayTime, hasSearchSubmitButton]);
 
   const handleInputChange = useCallback(
-    (
-      filterId: string,
-      value:
-        | string
-        | { label: string }
-        | null
-        | undefined
-        | (string | { label: string })[],
-      submit: boolean = false
-    ) => {
-      let extractedLabel: string | undefined = undefined;
-
-      if (Array.isArray(value)) {
-        // Extract the first valid string or { label: string } object from the array
-        const validValue = value.find(
-          (v) =>
-            typeof v === "string" || (typeof v === "object" && "label" in v)
-        );
-
-        if (validValue) {
-          extractedLabel =
-            typeof validValue === "string" ? validValue : validValue.label;
-        }
-      } else if (
-        value !== null &&
-        value !== undefined &&
-        (typeof value === "string" ||
-          (typeof value === "object" && "label" in value))
-      ) {
-        extractedLabel = typeof value === "string" ? value : value.label;
-      }
+    (filterId: string, value: DataFilterValue, submit: boolean = false) => {
+      setInputValues({ ...inputValues, [filterId]: value });
 
-      if (extractedLabel !== undefined) {
-        setInputValues({ ...inputValues, [filterId]: extractedLabel });
-
-        if (submit) {
-          const updatedFilters = filtersProp.map((filter) => ({
-            ...filter,
-            value:
-              filter.id === filterId ? extractedLabel : inputValues[filter.id],
-          }));
+      if (submit) {
+        const updatedFilters = filtersProp.map((filter) => ({
+          ...filter,
+          value: filter.id === filterId ? value : inputValues[filter.id],
+        }));
 
-          setFilters(updatedFilters);
-        }
+        setFilters(updatedFilters);
       }
     },
     [inputValues, filtersProp]
   );
 
+  const getAutoCompleteLabel = (
+    value:
+      | string
+      | { label: string }
+      | null
+      | undefined
+      | (string | { label: string })[]
+  ): string => {
+    if (Array.isArray(value)) {
+      // Iterating to find the label
+      for (const valueElement of value) {
```

Comment:
Is the value array ever more than one entry?

Reply:
I didn't explore this case but [multi-select is an option](https://odyssey-storybook.okta.design/iframe.html?id=mui-components-forms-autocomplete--docs&full=1&addons=0&stories=0&panel=false&nav=false&viewMode=docs#multiple) for `Autocomplete`

Reply:
So, it could theoretically be something like this?
```js
[{label: 'some label'}, 'another label']
```